mongoose/mongoose.d.ts

```diff
@@ -871,7 +871,7 @@ declare module "mongoose" {
     /** Hash containing current validation errors. */
     errors: Object;
     /** The string version of this documents _id. */
-    id: string;
+    id?: string;
```

Comment:
Just tried this an gave me an error because the id is not within an interface but rather inside `class MongooseDocument`.

A way around it (though not pretty so should probably be done differently) is to:

````
  interface OptianalId {
    /** The string version of this documents _id. */
    id?: string;
  }
  
  /*
   * section document.js
   * http://mongoosejs.com/docs/api.html#document-js
   */
  class MongooseDocument implements OptianalId 
```